staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd_test.go

```diff
@@ -160,6 +163,61 @@ func TestDelete(t *testing.T) {
 	test.TestDelete(validNewCustomResource())
 }
 
+func TestGenerationNumber(t *testing.T) {
+	// enable alpha feature CustomResourceSubresources
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CustomResourceSubresources, true)()
+
+	storage, server := newStorage(t)
+	defer server.Terminate(t)
+	defer storage.CustomResource.Store.DestroyFunc()
+	modifiedRno := *validNewCustomResource()
+	modifiedRno.SetGeneration(10)
+	ctx := genericapirequest.NewDefaultContext()
+	cr, err := createCustomResource(storage.CustomResource, modifiedRno, t)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	etcdCR, err := storage.CustomResource.Get(ctx, cr.GetName(), &metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	storedCR, _ := etcdCR.(*unstructured.Unstructured)
+
+	// Generation initialization
+	if storedCR.GetGeneration() != 1 {
+		t.Fatalf("Unexpected generation number %v", storedCR.GetGeneration())
+	}
+
+	// Updates to spec should increment the generation number
+	setSpecReplicas(storedCR, getSpecReplicas(storedCR)+1)
+	if _, _, err := storage.CustomResource.Update(ctx, storedCR.GetName(), rest.DefaultUpdatedObjectInfo(storedCR), rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	etcdCR, err = storage.CustomResource.Get(ctx, cr.GetName(), &metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	storedCR, _ = etcdCR.(*unstructured.Unstructured)
+	if storedCR.GetGeneration() != 2 {
+		t.Fatalf("Unexpected generation, spec: %v", storedCR.GetGeneration())
+	}
+
+	// Updates to status should not increment either spec or status generation numbers
```

Comment:
nit in the comment: status generation number doesn't exist anymore.

sorry for not catching before!

Reply:
Done. No problem,Thanks for your quick review.